@/components/ui/calendar.tsx

```diff
@@ -1,70 +1,153 @@
+      day_disabled: 'text-muted-foreground opacity-50',
+      day_range_middle:
+        'aria-selected:bg-accent aria-selected:text-accent-foreground',
+      day_hidden: 'invisible',
+      ...classNames,
+    }}
+    components={{
+      // eslint-disable-next-line consistent-return,react/no-unstable-nested-components
+      Dropdown: (props) => {
+        const { fromDate, fromMonth, fromYear, toDate, toMonth, toYear } =
+          useDayPicker();
+        const { goToMonth, currentMonth } = useNavigation();
+
+        if (props.name === 'months') {
+          const selectedItems = Array.from({ length: 12 }, (_, i) => ({
+            value: i.toString(),
+            label: format(setMonth(new Date(), i), 'MMMM', { locale }),
+          }));
+
+          return (
+            <Select
+              onValueChange={(newValue) => {
+                const newDate = new Date(currentMonth);
+                newDate.setMonth(parseInt(newValue, 10));
+                goToMonth(newDate);
+              }}
+            >
+              <SelectTrigger className="p-0 m-0 border-0 shadow-none outline-0">
+                {format(currentMonth, 'MMMM', { locale })}
+              </SelectTrigger>
+              <SelectContent>
+                {selectedItems.map((selectItem) => (
+                  <SelectItem value={selectItem.value}>
+                    {selectItem.label}
+                  </SelectItem>
+                ))}
+              </SelectContent>
+            </Select>
+          );
+        }
+        if (props.name === 'years') {
+          const earliestYear =
+            fromYear || fromMonth?.getFullYear() || fromDate?.getFullYear();
+          const latestYear =
+            toYear || toMonth?.getFullYear() || toDate?.getFullYear();
+
+          if (earliestYear && latestYear) {
+            const yearsLength = latestYear - earliestYear + 1;
+
+            const selectedItems = Array.from(
+              { length: yearsLength },
+              (_, i) => ({
+                value: (earliestYear + i).toString(),
+                label: (earliestYear + i).toString(),
+              })
+            );
+
+            return (
+              <Select
+                onValueChange={(newValue) => {
+                  const newDate = new Date(currentMonth);
+                  newDate.setFullYear(parseInt(newValue, 10));
+                  goToMonth(newDate);
+                }}
+                value={props.value?.toString()}
+              >
+                <SelectTrigger className="p-0 m-0 border-0 shadow-none outline-0">
+                  {currentMonth.getFullYear()}
+                </SelectTrigger>
+                <SelectContent>
+                  {selectedItems.map((selectItem) => (
+                    <SelectItem value={selectItem.value}>
+                      {selectItem.label}
+                    </SelectItem>
+                  ))}
+                </SelectContent>
+              </Select>
```

Comment:
this code is a duplicate of the previous one can we merge them together and have one return state for this component, I feel something like the next psudo code  world work:


```js
let itemsList = []
if (month) {
itemsList = doSomethingToFillIt() 
}
if (year) {
itemsList = fillit()
doSomething
}

return <SelectComponent>
```

Reply:
As for the setFullYear and seMonth, we can have a condition inside the onSelect Change. 
This will help us to maintain one select style for the months and years, if in the future wanted to change it we will have on place to edit.